src/timesync/test-timesync.c

```diff
@@ -13,16 +14,65 @@ TEST(manager_parse_string) {
 
         _cleanup_(manager_freep) Manager *m = NULL;
 
-        assert_se(manager_new(&m) == 0);
+        ASSERT_TRUE(manager_new(&m) == 0);
 
-        assert_se(!m->have_fallbacks);
-        assert_se(manager_parse_server_string(m, SERVER_FALLBACK, NTP_SERVERS) == 0);
-        assert_se(m->have_fallbacks);
-        assert_se(manager_parse_fallback_string(m, NTP_SERVERS) == 0);
+        ASSERT_TRUE(!m->have_fallbacks);
+        ASSERT_TRUE(manager_parse_server_string(m, SERVER_FALLBACK, NTP_SERVERS) == 0);
+        ASSERT_TRUE(m->have_fallbacks);
+        ASSERT_TRUE(manager_parse_fallback_string(m, NTP_SERVERS) == 0);
 
-        assert_se(manager_parse_server_string(m, SERVER_SYSTEM, "time1.foobar.com time2.foobar.com axrfav.,avf..ra 12345..123") == 0);
-        assert_se(manager_parse_server_string(m, SERVER_FALLBACK, "time1.foobar.com time2.foobar.com axrfav.,avf..ra 12345..123") == 0);
-        assert_se(manager_parse_server_string(m, SERVER_LINK, "time1.foobar.com time2.foobar.com axrfav.,avf..ra 12345..123") == 0);
+        ASSERT_TRUE(manager_parse_server_string(m, SERVER_SYSTEM, "time1.foobar.com time2.foobar.com axrfav.,avf..ra time2.foobar.com:1234 10.0.0.1 fe80::1 [fe80::1] 10.0.0.1:1234 [fe80::1]:1234 12345..123") == 0);
+        ASSERT_TRUE(manager_parse_server_string(m, SERVER_FALLBACK, "time1.foobar.com time2.foobar.com axrfav.,avf..ra time2.foobar.com:1234 10.0.0.1 fe80::1 [fe80::1] 10.0.0.1:1234 [fe80::1]:1234 12345..123") == 0);
+        ASSERT_TRUE(manager_parse_server_string(m, SERVER_LINK, "time1.foobar.com time2.foobar.com axrfav.,avf..ra time2.foobar.com:1234 10.0.0.1 fe80::1 [fe80::1] 10.0.0.1:1234 [fe80::1]:1234 12345..123") == 0);
+}
+
+TEST(server_name_parse_port) {
+        /* When supplied with a port number, use it (and re-use it if we fail to connect), and parse all of
+         * host.domain, host.domain:port, I.P.v.4, IP::v:6, [IP::v:6], I.P.v.4:port, [IP::v:6]:port appropriately. */
+        _cleanup_(server_name_freep) ServerName *n = NULL;
+        n = new(ServerName, 1);
+        *n = (ServerName) {
+                .string = strdupa_safe("time1.foobar.com"),
+                .overridden_port = NULL,
+        };
+        ASSERT_NULL(n->overridden_port); //no override
```

Comment:
as per coding style: we do not use // for explanatory comments, but use /* */ for that. 

// comments are left for FIXME style comments